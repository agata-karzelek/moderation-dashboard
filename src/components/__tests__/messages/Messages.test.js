import React from "react";
import { shallow } from "enzyme";
import Messages from "../../messages/Messages";

jest.mock("react-router", () => ({
  useLocation: jest.fn().mockReturnValue({
    pathname: "/channels/messages",
    search: "",
    hash: "",
    state: {
      channel: "pro.one",
    },
    key: "bi8m6y",
  }),
}));

describe("Test Case For Messages", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<Messages channelName="pro.one" />);
  });

  test("Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
