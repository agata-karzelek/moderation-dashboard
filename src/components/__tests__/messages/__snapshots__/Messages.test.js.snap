// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Case For Messages Snapshot 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Messages"
  />
  <WithStyles(ForwardRef(Grid))
    container={true}
    justify="space-between"
    spacing={1}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      sm={9}
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
        justify="space-between"
      >
        <WithStyles(ForwardRef(Grid))
          className="makeStyles-header-2"
          item={true}
          sm={12}
        >
          <WithStyles(ForwardRef(Grid))
            container={true}
            justify="flex-start"
          >
            <Styled(MuiBox)
              mr={1}
            >
              <WithStyles(ForwardRef(Avatar))
                className="makeStyles-avatar-6"
                variant="square"
              >
                PR
              </WithStyles(ForwardRef(Avatar))>
            </Styled(MuiBox)>
            <WithStyles(ForwardRef(Grid))
              item={true}
              md={8}
              sm={8}
              xs={12}
            >
              <small
                className="makeStyles-values-5"
              >
                pro.one
                 (
                )
              </small>
              <br />
              <small
                className="makeStyles-appName-7"
              />
            </WithStyles(ForwardRef(Grid))>
            <WithStyles(ForwardRef(Grid))
              item={true}
              lg={3}
              md={3}
              sm={3}
            >
              <WithStyles(ForwardRef(Grid))
                container={true}
                justify="space-between"
              >
                <WithStyles(ForwardRef(Grid))
                  item={true}
                />
                <WithStyles(ForwardRef(Grid))
                  item={true}
                >
                  <Styled(MuiBox)
                    mt={1}
                  >
                    <ToggleChat
                      setToggledVal={[Function]}
                      toggledVal="chat"
                    />
                  </Styled(MuiBox)>
                </WithStyles(ForwardRef(Grid))>
              </WithStyles(ForwardRef(Grid))>
            </WithStyles(ForwardRef(Grid))>
          </WithStyles(ForwardRef(Grid))>
          <Styled(MuiBox)
            mt={1}
          >
            <WithStyles(ForwardRef(Grid))
              container={true}
              justify="center"
            >
              <WithStyles(ForwardRef(Button))
                className="makeStyles-messagesCount-19"
                disabled={true}
              >
                <MessagesCount
                  channelName="pro.one"
                />
              </WithStyles(ForwardRef(Button))>
            </WithStyles(ForwardRef(Grid))>
          </Styled(MuiBox)>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm={12}
      >
        <Chat
          channel="pro.one"
          isLoading={false}
          messageToEdit={Array []}
          messages={Array []}
          setMessageToEdit={[Function]}
          toggledVal="chat"
          updated={[Function]}
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-channelMembers-13"
      item={true}
      sm={3}
    >
      <Members
        channelMembers={false}
        channelName="pro.one"
        page=""
        selectedMemberDetail={[Function]}
        totalCount={0}
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</Fragment>
`;
